compiler/rustc_const_eval/src/interpret/intrinsics.rs

```diff
@@ -697,4 +747,54 @@ impl<'tcx, M: Machine<'tcx>> InterpCx<'tcx, M> {
         let rhs_bytes = get_bytes(self, rhs)?;
         interp_ok(Scalar::from_bool(lhs_bytes == rhs_bytes))
     }
+
+    fn float_bin_op_intrinsic<F>(
+        &mut self,
+        op: FloatBinOp,
+        args: &[OpTy<'tcx, M::Provenance>],
+        dest: &MPlaceTy<'tcx, M::Provenance>,
+    ) -> InterpResult<'tcx, ()>
+    where
+        F: rustc_apfloat::Float + rustc_apfloat::FloatConvert<F> + Into<Scalar<M::Provenance>>,
+    {
+        let a: F = self.read_scalar(&args[0])?.to_float()?;
+        let b: F = self.read_scalar(&args[1])?.to_float()?;
+        let res = match op {
+            FloatBinOp::Min => adjust_nan(self, a.min(b), &[a, b]),
+            FloatBinOp::Max => adjust_nan(self, a.max(b), &[a, b]),
+            FloatBinOp::Copysign => a.copy_sign(b), // bitwise, no NaN adjustments
+        };
+        self.write_scalar(res, dest)?;
+        interp_ok(())
+    }
+
+    fn float_abs_intrinsic<F>(
+        &mut self,
+        args: &[OpTy<'tcx, M::Provenance>],
+        dest: &MPlaceTy<'tcx, M::Provenance>,
+    ) -> InterpResult<'tcx, ()>
+    where
+        F: rustc_apfloat::Float + rustc_apfloat::FloatConvert<F> + Into<Scalar<M::Provenance>>,
+    {
+        let x: F = self.read_scalar(&args[0])?.to_float()?;
+        // bitwise, no NaN adjustments
+        self.write_scalar(x.abs(), dest)?;
+        interp_ok(())
+    }
+}
+
+enum FloatBinOp {
```

Comment:
```suggestion
enum FloatBinIntrinsic {
```
We use `BinOp` for the MIR binops, IMO we shouldn't mix up that terminology.

Also please move this above the functions that use it.